app/queues/dns/delete-record-flow.server.ts

```diff
@@ -0,0 +1,89 @@
+import { FlowProducer } from 'bullmq';
+import { redis } from '~/lib/redis.server';
+import { buildDomain } from '~/utils';
+import { dnsUpdateQueueName } from './workers/dns-update-worker.server';
+import { checkDnsStatusQueueName } from './workers/poll-dns-status-worker.server';
+import { syncDbStatusQueueName } from './workers/sync-db-status-worker.server';
+import { WorkType } from './add-record-flow.server';
+
+import type { FlowJob } from 'bullmq';
+import type { Record } from '@prisma/client';
+import type { DnsUpdaterData } from './workers/dns-update-worker.server';
+import type { DbRecordSynchronizerData } from './workers/sync-db-status-worker.server';
+import type { Name } from './add-record-flow.server';
+
+interface BaseDeleteDnsRequestData {
```

Comment:
Instead of needing a name like this, you can do it all at once:

```ts
export type DeleteDnsRequestData = {
  id: Record['id'];
  username: Record['username'];
  type: Record['type'];
  value: Record['value'];
} & Name;
```

Reply:
It's been so great for me to learn all these typing techniques. Thank you